src/libraries/Common/tests/SourceGenerators/RoslynTestUtils.cs

```diff
@@ -0,0 +1,300 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using Xunit;
+
+namespace SourceGenerators.Tests
+{
+    internal static class RoslynTestUtils
+    {
+        /// <summary>
+        /// Creates a canonical Roslyn project for testing.
+        /// </summary>
+        /// <param name="references">Assembly references to include in the project.</param>
+        /// <param name="includeBaseReferences">Whether to include references to the BCL assemblies.</param>
+        public static Project CreateTestProject(IEnumerable<Assembly>? references, bool includeBaseReferences = true)
+        {
+#pragma warning disable SA1009 // Closing parenthesis should be spaced correctly
```

Comment:
I don't understand why this needs to be disabled here. Also, I don't think SA1009 is a warning or error:
[https://github.com/dotnet/runtime/blob/184a5b103ca9fec1e7e7e0d3a411a6d15f1bb867/eng/CodeAnalysis.ruleset#L269](https://github.com/dotnet/runtime/blob/184a5b103ca9fec1e7e7e0d3a411a6d15f1bb867/eng/CodeAnalysis.ruleset#L269)